packages/switch/src/index.tsx

```diff
@@ -20,11 +20,17 @@ function useSwitch(
   const [current, setCurrent] = React.useState(controlledValue ?? defaultValue)
   useChange(current, onChange)
 
+  React.useEffect(() => {
+    if (!isBoolean(controlledValue) || controlledValue === current) return
```

Comment:
This can be `typeof controlledValue === 'boolean'` and you can get rid of the `isBoolean` function